tests/tracing/test_sampling.py

```diff
@@ -284,3 +284,61 @@ def test_warns_and_sets_sampled_to_false_on_invalid_traces_sampler_return_value(
         transaction = start_transaction(name="dogpark")
         logger.warning.assert_any_call(StringContaining("Given sample rate is invalid"))
         assert transaction.sampled is False
+
+
+@pytest.mark.parametrize(
+    "traces_sample_rate,sampled_output,reports_output",
+    [
+        (None, False, []),
+        (0.0, False, [("sample_rate", "transaction")]),
+        (1.0, True, []),
```

Comment:
My curiosity, were there no tests covering this case before?

The key behavior this PR is changing is the first case, when `traces_sample_rate` is `None`.

I read the test names again, and while it is better now that each input and output is an explicit parameter, it is not so obvious why only the `0.0-False-...` case has a report (one needs to think/know what reports are for reporting "lost" events).

Maybe a clearer test would be one where there's always some lost event, and the expected output for the `None-...` is that only the transaction-related report should be skipped, but not one about errors. Is that covered somewhere else?

Reply:
no that's not covered afaik, will add one.

One high level comment I want to make here re: my personal testing philosophy is about the impossibility of testing all cases.
If we have 20 config parameters for `sentry.init` each taking 2 values (boolean), all possible combinations of those are `2^20 =1048576` total cases. Even more if those are not booleans. If any of those values is continuous, theoretically you need an infinite number of tests to test every possible state of your program/api surface.
Given this theoretical state of affairs, as _engineers_, we need to pick and choose where we write tests and not all tests add value/are worth adding.

I will add this one anyway but I just wanted to clarify how I think about testing for the future.

Reply:
I see your extrapolation. But we're not trying to test all combinations of input, but rather behaviors, and there are not so many interesting behaviors to test here.

For testing arbitrary inputs then we can rely on other tools such as fuzzing, and not hand-written unit tests.

Reply:
> Maybe a clearer test would be one where there's always some lost event, and the expected output for the None-... is that only the transaction-related report should be skipped, 

this tbh is a fairly contrived case which involves `sample_rate` too so I don't believe this belongs to the scope that these tests are trying to tackle.